src/main/java/cz/student/adambrg/UserDao.java

```diff
@@ -0,0 +1,139 @@
+package cz.student.adambrg;
+
+import cz.common.DbUtil;
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+    public static final String CREATE_USER =
+            "INSERT INTO users(username, email, password) VALUES (?, ?, ? )";
+
+
+    public User create(User user) { //metoda pro vytvareni uzivatele
+        try (Connection conn = DbUtil.getConnection()) {
+            PreparedStatement statement = conn.prepareStatement(CREATE_USER, PreparedStatement.RETURN_GENERATED_KEYS);
+
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.executeUpdate();
+
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getInt(1));
+                System.out.println("User ID " + user.getId());
+
+            }
+            return user;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("ERROR while creating a user");
+            return null;
+
+        }
+    }
+
+    public static final String UPDATE_USER =
+            "UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?";
+
+    public void update(User user, Connection conn) { //metoda pro aktualizovani inforamci uzivatele
+        try {
+            PreparedStatement statement = conn.prepareStatement(UPDATE_USER);
+
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.setInt(4, user.getId());
+
+            statement.executeUpdate();
+            System.out.println("Changes applied to user with ID: " + user.getId());
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("ERROR while updating the user");
```

Comment:
BUG: je treba overit jestli opravdu doslo k aktualizaci uzivatelu.


```suggestion
 final int affectedRows = statement .executeUpdate();

            if (affectedRows > 0) {
                System.out.println("User with id " + user.getId() + " updated successfully.");
            } else {
                System.out.println("User update failed. No rows affected.");
            }
```

Reply:
opraveno